internal/cfgmgr/cfgmgr.go

```diff
@@ -0,0 +1,163 @@
+package cfgmgr
+
+import (
+	"fmt"
+	"strings"
+	"unicode/utf16"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"github.com/pkg/errors"
+)
+
+//go:generate go run ../../mksyscall_windows.go -output zsyscall_windows.go cfgmgr.go
+
+//sys cmGetDeviceIDListSize(pulLen *uint32, pszFilter *byte, uFlags uint32) (hr error) = cfgmgr32.CM_Get_Device_ID_List_SizeA
+//sys cmGetDeviceIDList(pszFilter *byte, buffer *byte, bufferLen uint32, uFlags uint32) (hr error)= cfgmgr32.CM_Get_Device_ID_ListA
+//sys cmLocateDevNode(pdnDevInst *uint32, pDeviceID string, uFlags uint32) (hr error) = cfgmgr32.CM_Locate_DevNodeW
+//sys cmGetDevNodeProperty(dnDevInst uint32, propertyKey *devPropKey, propertyType *uint32, propertyBuffer *uint16, propertyBufferSize *uint32, uFlags uint32) (hr error) = cfgmgr32.CM_Get_DevNode_PropertyW
+
+type devPropKey struct {
+	fmtid guid.GUID
+	pid   uint32
+}
+
+const (
+	_CM_GETIDLIST_FILTER_BUSRELATIONS = uint32(0x00000020)
+
+	_CM_LOCATE_DEVNODE_NORMAL = uint32(0x00000000)
+
+	_DEVPROP_TYPE_STRING        = uint32(0x00000012)
+	_DEVPROP_TYPEMOD_LIST       = uint32(0x00002000)
+	_DEVPROP_TYPE_STRING_LIST   = uint32(_DEVPROP_TYPE_STRING | _DEVPROP_TYPEMOD_LIST)
+	_DEVPKEY_LOCATIONPATHS_GUID = "a45c254e-df1c-4efd-8020-67d146a850e0"
+)
+
+func getDevPKeyDeviceLocationPaths() (*devPropKey, error) {
+	guid, err := guid.FromString(_DEVPKEY_LOCATIONPATHS_GUID)
+	if err != nil {
+		return nil, err
+	}
+	return &devPropKey{
```

Comment:
It would be good to have more information here on what exactly is going on. I looked briefly at the docs and it's not clear if these specific DEVPROPKEY values come from. In general I would try to put yourself in the shoes of someone who has no familiarity with this area, and ask yourself what information you need to understand the basics of how this code works. Then we should make sure that's covered by comments and links to docs, as much as possible.

Reply:
I'm adding a comment to make it more clear. But just adding here too: this function creates a DEVPROPKEY struct by using the contents supplied to [DEFINE_DEVPROPKEY](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/define-devpropkey) in devpkey.h for the DEVPKEY_Device_LocationPaths property. 

```
DEFINE_DEVPROPKEY(DEVPKEY_Device_LocationPaths,          0xa45c254e, 0xdf1c, 0x4efd, 0x80, 0x20, 0x67, 0xd1, 0x46, 0xa8, 0x50, 0xe0, 37);    // DEVPROP_TYPE_STRING_LIST
```